backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +63,128 @@ private void updatePersonFacility(Person person, UUID facilityId) {
     person.setFacility(facility);
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            false,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // For readablility below. Get the OrgId for the currently logged in user.
+  // don't trust an id provided by the user.
+  private UUID getOrgId() {
+    return _os.getCurrentOrganization().getInternalId();
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), false);
+  // Specifications filters for queries
+  private Specification<Person> inWholeOrganizationFilter() {
+    return (root, query, cb) ->
+        cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId());
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getAllPatients(int pageOffset, int pageSize) {
-    return _repo
-        .findAllByOrganization(
-            _os.getCurrentOrganization(), false, PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // Note: Patients with NULL facilityIds appear in ALL facilities.
+  private Specification<Person> inFacilityFilter(@NotNull UUID facilityId) {
+    return (root, query, cb) ->
+        cb.and(
+            cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId()),
+            cb.or(
+                cb.isNull(root.get(SpecField.Facility)), // null check first
+                cb.equal(root.get(SpecField.Facility).get(SpecField.InternalId), facilityId)));
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getAllPatientsCount() {
-    return _repo.countAllByOrganization(_os.getCurrentOrganization(), false);
+  private Specification<Person> nameMatchesFilter(
+      @Size(min = MINIMUMCHARFORSEARCH) String namePrefixMatch) {
+    String likeString = namePrefixMatch.trim().toLowerCase() + "%";
+    return (root, query, cb) ->
+        cb.or(
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.FirstName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.MiddleName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.LastName)), likeString));
```

Comment:
@benwarfield-usds Part of this PR should probably be the addition of an index on the `person` table? We could wait and see what the performance is or we can preemptively make one.

When we apply the index, the servers might take a while to initially create the index? Should we plan for it to take a lot longer? (just brainstorming) Are there any concerns with liquibase taking a really long time and something else timing out?

Reply:
I wouldn't worry about the index taking too long—we've still got mere thousands of patient records AFAIK.

I would not make the index part of this PR unless we test this out in staging and it's awwwful performance—figuring out which indexes do and don't help is actually a bit of a guessing game, and we shouldn't delay this feature unless we have to.

(That said: I would bet heavily that indexing on `organization_id` alone is sufficient to make this pretty performant until we get up to state-university-scale orgs.)

Reply:
To overextend this conversation a bit, there are three (four?) possible index plans:
* index on organization_id alone
* index on organization_id and facility_id (bearing in mind that null is not equal to null, so sometimes you have to do weird things if you want to fully index a nullable field)
* indexes on
  * organization_id and first_name
  * organization_id and middle_name
  * organization_id and last_name
* alternatively and probably preferably, indexes on
   * organization
   * first_name
   * last_name
   * middle_name

I strongly suspect that the difference in selectivity between the first and second indexes is not worth the trouble, and the postgresql optimizer is likely smart enough to use the last scenario at least as well as the (rather pricey) third scenario, so that leaves us with either one really obvious index or three or four indexes, which include the obvious one as part of the build-out. This being the case, I would advocate for an empirical approach to which indexes we actually add. 😸

Reply:
Agreed. The queries always include an organization_id so the execution plan likely restricts on that term first. So, as long as we don't have thousands of facilities for a single org, it should be ok to not index on facilities.